src/server/JasmineGraphInstanceService.cpp

```diff
@@ -2692,3 +2725,319 @@ map<long, long> JasmineGraphInstanceService::calculateLocalOutDegreeDistribution
+                            bzero(data, 301);
+                            read(sockfd, data, 300);
+                            response = (data);
+
+                            if (response.compare(JasmineGraphInstanceProtocol::BATCH_UPLOAD_WAIT) == 0) {
+                                instance_logger.log("Received : " + JasmineGraphInstanceProtocol::BATCH_UPLOAD_WAIT, "info");
+                                sleep(1);
+                                continue;
+                            } else if (response.compare(JasmineGraphInstanceProtocol::BATCH_UPLOAD_ACK) == 0) {
+                                instance_logger.log("Received : " + JasmineGraphInstanceProtocol::BATCH_UPLOAD_ACK, "info");
+                                instance_logger.log("Batch upload completed", "info");
+                                break;
+                            }
+                        }
+                    }
+                } else {
+                    instance_logger.log("There was an error in the upload process and the response is :: " + response, "error");
+                }
+                close(sockfd);
+            }
+        }
+    }
+    return 0;
+}
+
+bool sendFileThroughService(std::string host, int dataPort, std::string fileName,
+                            std::string filePath, std::string masterIP) {
+    Utils utils;
+    int sockfd;
+    char data[300];
+    socklen_t len;
+    struct sockaddr_in serv_addr;
+    struct hostent *server;
+
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+    if (sockfd < 0) {
+        std::cerr << "Cannot accept connection" << std::endl;
+        return 0;
+    }
+
+    server = gethostbyname(host.c_str());
+    if (server == NULL) {
+        std::cerr << "ERROR, no host named " << server << std::endl;
+        exit(0);
+    }
+
+    bzero((char *) &serv_addr, sizeof(serv_addr));
+    serv_addr.sin_family = AF_INET;
+    bcopy((char *) server->h_addr,
+          (char *) &serv_addr.sin_addr.s_addr,
+          server->h_length);
+    serv_addr.sin_port = htons(dataPort);
+    if (connect(sockfd, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
+        std::cerr << "ERROR connecting to port " << dataPort << std::endl;
+    }
+
+    int result_wr = write(sockfd, fileName.c_str(), fileName.size());
+
+    if(result_wr < 0) {
+        instance_logger.log("Error writing to socket", "error");
+    }
+
+    bzero(data, 301);
+    read(sockfd, data, 300);
+    string response = (data);
+    response = utils.trim_copy(response, " \f\n\r\t\v");
+
+    if (response.compare(JasmineGraphInstanceProtocol::SEND_FILE) == 0) {
+        std::cout << "Sending file " << filePath << " through port " << dataPort << std::endl;
+
+        FILE *fp = fopen(filePath.c_str(), "r");
+        if (fp == NULL) {
+            instance_logger.log("Error opening file", "error");
+            close(sockfd);
+            return 0;
+        }
+
+        for (;;) {
+            unsigned char buff[1024] = {0};
```

Comment:
Instead of 1024 can we have a meaningful constant here?

Reply:
Added a constant

Reply:
It seems `INSTANCE_LONG_DATA_LENGTH` constant serves the same purpose. Shall we use that instead of defining this new constant?

Reply:
A new constant was added as this was related to file transfer and INSTANCE_LONG_DATA_LENGTH has been used generally for message passing